specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/codeContainer.json

```diff
@@ -0,0 +1,324 @@
+            "schema": {
+              "$ref": "#/definitions/CodeContainerResourceRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeContainerResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/CodeContainerResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "CodeContainerNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Code Container identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "CodeContainerResource": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The resource URL of the entity (not URL encoded).",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the resource entity.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The resource provider and type.",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/CodeContainer"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "additionalProperties": false,
+      "description": "Azure Resource Manager resource Envelope"
+    },
+    "CodeContainerResourcePaginatedResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CodeContainerResource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "CodeContainer": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The asset container description text."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The asset tag dictionary. Tags can be added, removed, and updated."
+        },
+        "latestVersions": {
+          "type": "array",
+          "items": {
+            "$ref": "codeVersion.json#/definitions/CodeVersionResource"
+          },
+          "description": "Latest Versions for all assets"
+        },
+        "name": {
```

Comment:
The inner "name" property on both CodeContainer and CodeVersion needs to be removed or renamed.  You shouldn't re-use top level property names in your property bag

Reply:
should these sort of duplicates show up in the top-most level field or the bottom-most level field?

Reply:
"name" is a required top level property on every resource.  It should only exist there

Reply:
some "name" properties were removed. The other name properties belong, but were not labeled as an azure-resource correctly. Everything _should_ be good now :)